mw-webapp/src/pages/authPage/AuthPage.tsx

```diff
@@ -1,14 +1,62 @@
-import {Link} from "react-router-dom";
+import {User, getRedirectResult, onAuthStateChanged} from "firebase/auth";
+import {useEffect, useState} from "react";
+import {Link, useNavigate} from "react-router-dom";
 import {useAuth} from "src/utils/useAuth";
 
 export const AuthPage = () => {
-  const handleGoogleSignIn = useAuth();
+  const {handleGoogleSignIn, handleLogout, writeNewUserData, auth} = useAuth();
+  const navigate = useNavigate();
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const handleAuthState = onAuthStateChanged(auth, (currentUser) => {
+      if (currentUser) {
+        setUser(currentUser);
+      } else {
+        setUser(null);
+      }
+    });
+    return () => {
+      handleAuthState();
+    };
+  }, []);
+
```

Comment:
I think better to move out the function and then just call it inside useEffect

Reply:
Done